### PR TITLE
feat: Rest of P0 features

### PR DESCRIPTION
**Pull request by Naman-Saxena1**
# Added Rest of P0 features
1. Genre Page to show all quizzes available for a specific genre
2. Quiz Rules Page 
3. Quiz Page to attempt quiz
4. Results to show correct options and final score

# Files to review
1. Quiz Page.tsx
- This page represent individual quiz page

2. QuizQuestion.tsx
- This component represents Each question(question no., question and options) in a quiz

# Login creds to test:
Email - newuser1@gmail.com
Password - zxcv

### Or you can create your own account from signup page

# Screenshots
Capture4
Capture5

### CODE REVIEW

File: src/components/QuizQuestion/QuizQuestion.tsx [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+            options
+        },
+        questionNumber,
+        setSelectedOptionResult,
+        isQuizSubmitted
+    } = props
+    const [isOptionSelected, setIsOptionSelected] = useState([false, false, false])
+
+    const selectQuestionOption = (event:React.MouseEvent, option:{optionValue: string,isCorrectOption: boolean}, optionIndex:number) => {
+        setSelectedOptionResult((prevResponseArray:Boolean[]):Boolean[]=>{
```
Comment (iRohitGaur) [inline src/components/QuizQuestion/QuizQuestion.tsx:20]: suggestion: use `boolean` here.

Read this for more info: https://www.tutorialsteacher.com/typescript/typescript-boolean

**Review by iRohitGaur — APPROVED** [review]
nice work!

Gave some suggestions, please look into it.

Good to merge with updates 👍🏻

**Review by darshandamre — APPROVED** [review]
all good, 2 suggestions
1. use prettier (and I have configured indentation to 2 spaces)
2. this might be a personal preference, but i like to keep my types in its respective component file/folder itself. recently, I have started following this principle `Things that change together should be located as close as reasonable." [Dan Abramov](https://twitter.com/dan_abramov)`. read this  [Colocation](https://kentcdodds.com/blog/colocation)

File: src/components/QuizQuestion/QuizQuestion.tsx (hunk 1 above)
Comment (Naman-Saxena1) [reply src/components/QuizQuestion/QuizQuestion.tsx:20]: TypeScript was giving me error when I tried boolean, hence I did this 😄

**Pull request merged**